src/util/linear_scan.rs

```diff
@@ -5,6 +5,9 @@ use crate::vm::ObjectModel;
 use crate::vm::VMBinding;
 use std::marker::PhantomData;
 
+// FIXME: This linear scanning algorithm is only used by MarkCompact.  It should use a local
```

Comment:
Currently linear scan can be used by any plan or by the binding (as a utility/debug tool) as long as vo bit is turned on. I don't know if it will be useful or not. Maybe we can engineer the linear scan code to take a side metadata spec as an argument (or const generic).

But you are right --  mark compact probably should use its own local metadata rather than the VO bit.

Malloc mark sweep also uses the vo bit. But I remember that we have some difficulties of using local metadata for it. We have to use the global vo/alloc bit in malloc mark sweep.

Reply:
I rephrased this comment a little bit.

Yes.  Malloc mark sweep and MarkCompact are the two cases which I think are misuses of the VO bit metadata.  As for malloc space, it needs a global metadata that has different semantics than the VO bit.  For example, if the user explicitly clears the VO bit to indicate that "the object's type metadata is destroyed and the GC shouldn't scan it or it will crash", malloc still needs a way to know it has an allocation at that address so that it can free it.  We may consider adding another per-16-bit (granularity of malloc) global metadata, and we may consider making the VO bit a tri-state metadata where the third state means "there is an allocation, but is no longer valid".